subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestSuitesIntegrationTest.groovy

```diff
@@ -0,0 +1,331 @@
+                doLast {
+                    assert integTest.testFramework instanceof ${JUnitPlatformTestFramework.canonicalName}
+                    assert configurations.integTestRuntimeClasspath.files.size() == 8
+                    assert configurations.integTestRuntimeClasspath.files.any { it.name == "junit-jupiter-5.7.1.jar" }
+                }
+            }
+        """
+        expect:
+        succeeds("checkConfiguration")
+    }
+
+    def "configuring test framework on custom test suite is honored in task and dependencies with #testingFrameworkDeclaration"() {
+        buildFile << """
+            plugins {
+                id 'java'
+            }
+            
+            repositories {
+                ${mavenCentralRepository()}
+            }
+
+            testing {
+                suites {
+                    integTest(JvmTestSuite) {
+                        ${testingFrameworkDeclaration}
+                    }
+                }
+            }
+
+            task checkConfiguration {
+                dependsOn integTest
+                doLast {
+                    assert integTest.testFramework instanceof ${testingFrameworkType.canonicalName}
+                    assert configurations.integTestRuntimeClasspath.files.any { it.name == "${testingFrameworkDep}" }
+                }
+            }
+        """
+        expect:
+        succeeds("checkConfiguration")
+
+        where:
+        testingFrameworkDeclaration | testingFrameworkType       | testingFrameworkDep
+        'useJUnit()'                | JUnitTestFramework         | "junit-4.13.jar"
+        'useJUnitJupiter()'         | JUnitPlatformTestFramework | "junit-jupiter-5.7.1.jar"
+        'useJUnit("4.12")'          | JUnitTestFramework         | "junit-4.12.jar"
+        'useJUnitJupiter("5.7.2")'  | JUnitPlatformTestFramework | "junit-jupiter-5.7.2.jar"
+    }
+
+    def "can override previously configured test framework on a test suite"() {
+        buildFile << """
+            plugins {
+                id 'java'
+            }
+            
+            repositories {
+                ${mavenCentralRepository()}
+            }
+
+            testing {
+                suites {
+                    integTest(JvmTestSuite) {
+                        useJUnit()
+                        useJUnitJupiter()
+                    }
+                }
+            }
+
+            task checkConfiguration {
+                dependsOn integTest
+                doLast {
+                    assert integTest.testFramework instanceof ${JUnitPlatformTestFramework.canonicalName}
+                    assert configurations.integTestRuntimeClasspath.files.size() == 8
+                    assert configurations.integTestRuntimeClasspath.files.any { it.name == "junit-jupiter-5.7.1.jar" }
+                }
+            }
+        """
+        expect:
+        succeeds("checkConfiguration")
+    }
+
+    def "task configuration overrules test suite configuration"() {
```

Comment:
To be extra-paranoid, perhaps add another version of this test that explicitly set the suite to use a non-default framework, then overrides the test to use a different framework.

Reply:
that's a good idea